src/deno_transport.ts

```diff
@@ -192,51 +178,87 @@ export class DenoTransport implements Transport {
     this._closed(reason).then().catch();
   }
 
-  private enqueue(frame: Uint8Array): Promise<void> {
-    if (this.done) {
+  flusher(): Promise<void> {
+    const data = this.frames.shift();
+    if (!data) {
       return Promise.resolve();
     }
-    const d = deferred<void>();
-    this.sendQueue.push({ frame, d });
-    if (this.sendQueue.length === 1) {
-      this.dequeue();
-    }
-    return d;
-  }
 
-  private dequeue(): void {
-    const [entry] = this.sendQueue;
-    if (!entry) return;
-    if (this.done) return;
-    const { frame, d } = entry;
-    write(frame, this.writer)
+    const d = deferred<void>();
+    writeAll(this.conn, data)
       .then(() => {
         if (this.options.debug) {
-          console.info(`< ${render(frame)}`);
+          console.info(`< ${render(data)}`);
         }
         d.resolve();
       })
       .catch((err) => {
         if (this.options.debug) {
-          console.error(`!!! ${render(frame)}: ${err}`);
+          console.error(`!!! ${render(data)}: ${err}`);
         }
         d.reject(err);
       })
       .finally(() => {
-        this.sendQueue.shift();
         this.dequeue();
       });
+
+    return d;
   }
 
-  send(frame: Uint8Array): void {
-    const p = this.enqueue(frame);
-    p.catch((_err) => {
-      // we ignore write errors because client will
-      // fail on a read or when the heartbeat timer
-      // detects a stale connection
+  private dequeue(d?: Deferred<void>): void {
+    // if we don't have data, do nothing
+    if (this.frames.length === 0) {
+      return;
+    }
+
+    // if we have a timer, we reached a limit of microtasks
+    // and we are waiting on a timer
+    if (this.syncTimer) {
+      // do nothing, there's a setTimeoutScheduled
+      return;
+    }
+    // if we have max number of pending microtasks
+    // schedule a timeout
+    if (this.microtasks === 1000) {
```

Comment:
Nit: Add a const for this, tested up to a million and doesn't fail.

```suggestion
    if (this.microtasks === 1000) {
```